internal/controllers/shared_vehicles.go

```diff
@@ -0,0 +1,85 @@
+package controllers
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/DIMO-Network/vehicle-events-api/internal/config"
+	"io"
+	"net/http"
+
+	"github.com/rs/zerolog"
+)
+
+type Vehicle struct {
+	TokenID string `json:"tokenId"`
+}
+
+func GetSharedVehicles(devLicense []byte, logger zerolog.Logger) ([]Vehicle, error) {
+	ethAddress := fmt.Sprintf("0x%x", devLicense)
+
+	identityAPIURL := config.GetSettings().IdentityAPIURL
+
+	query := `
+		query($eth: String!) {
+			vehicles(first: 50, filterBy: { privileged: $eth }) {
+				nodes {
+					tokenId
+				}
+			}
+		}`
+	payload := map[string]interface{}{
+		"query": query,
+		"variables": map[string]interface{}{
+			"eth": ethAddress,
+		},
+	}
+	payloadBytes, err := json.Marshal(payload)
+	if err != nil {
+		logger.Error().Err(err).Msg("failed to marshal GraphQL payload")
+		return nil, err
+	}
+
+	resp, err := http.Post(identityAPIURL, "application/json", bytes.NewBuffer(payloadBytes))
+	if err != nil {
+		logger.Error().Err(err).Msg("failed to POST to identity API")
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		logger.Error().Err(err).Msg("failed to read response body")
+		return nil, err
+	}
+
+	if resp.StatusCode < 200 || resp.StatusCode > 299 {
+		errMsg := fmt.Sprintf("identity API returned non-2xx status: %d", resp.StatusCode)
+		logger.Error().Msg(errMsg)
+		return nil, errors.New(errMsg)
```

Comment:
```suggestion
		return nil, fmt.Errorf("identity API returned non-2xx status: %d", resp.StatusCode)
```
Also, do you need to log here or should the caller to your function `GetSharedVehicles` do the logging? You typically want the thing highest in the stack to do the logging